### PR TITLE
Core CDEvents to render as CloudEvent and schema validation 

### PR DESCRIPTION
**Pull request by rjalander**
This PR is a continuation of https://github.com/cdevents/sdk-java/pull/35, to implement other Core events

### CODE REVIEW

**Review by afrittoli — APPROVED** [review]
Thank you!! This looks good.
Do we have anything to remove from the old implementation?

**Comment by rjalander** [comment]
Thank you.
Yes, we need to remove the old implementation [CDEventTypes.java](https://github.com/cdevents/sdk-java/blob/main/src/main/java/dev/cdevents/CDEventTypes.java) and it's UT related files.

**Comment by afrittoli** [comment]
> Thank you. Yes, we need to remove the old implementation [CDEventTypes.java](https://github.com/cdevents/sdk-java/blob/main/src/main/java/dev/cdevents/CDEventTypes.java) and it's UT related files.

It would be nice to have it in this PR, I think, or do you plan on removing the old implementation all the end at once?
Since this is a new package it may be ok, as long is all happen this week.

**Comment by rjalander** [comment]
Yes, we are still creating PRs for Source Code, Continues Integration and Deployment events to implement with this approach and planning to complete in couple of days.
I think I will create another PR once at the end.

File: src/test/java/dev/cdevents/CDEventsTest.java [hunk 1]
```diff
@@ -22,7 +22,7 @@ void createPipelineRunFinishedEventAsCloudEvent() {
         cdEvent.setSource(URI.create("http://dev.cdevents"));
 
         cdEvent.setSubjectId("/dev/pipeline/run/subject");
-        cdEvent.setSubjectSource(URI.create("/dev/pipeline/run/subject"));
+        cdEvent.setSubjectSource(URI.create("http://dev/pipeline/run/subject"));
```
Comment (afrittoli) [inline src/test/java/dev/cdevents/CDEventsTest.java:25]: NIT: why `http://`? It's fine, but even the string starting with `/dev` should be a valid URI.

**Comment by afrittoli** [comment]
Thank you!
It just came to mind that perhaps we should use the spec version v0.1.2 instead of v0.1.0 as starting point here since there were a couple of issues in the v0.1.0 version, the reason why we made the patch releases.

**Comment by rjalander** [comment]
Ok, so we don't need to maintain a v0.1.0 version for Java-SDK and I can update this and other event PRs to use v0.1.2 from now?

**Comment by afrittoli** [comment]
> Ok, so we don't need to maintain a v0.1.0 version for Java-SDK and I can update this and other event PRs to use v0.1.2 from now?

Indeed, please use v0.1.2 - there are a few [bug fixes](https://github.com/cdevents/spec/compare/v0.1.0...v0.1.2) that's good to have in.

**Comment by rjalander** [comment]
Thank you @afrittoli, updated this PR to use 0.1.2 specification.

**Comment by rjalander** [comment]
@afrittoli are we good to merge this PR now.

**Review by afrittoli — APPROVED** [review]

**Pull request merged**